app/models/provider/synth.rb

```diff
@@ -0,0 +1,59 @@
+class Provider::Synth < Provider::Base
+  def initialize(api_key)
+    @api_key = api_key
+  end
+
+  def fetch_exchange_rate(from:, to:, date:)
+    response = Faraday.get("#{base_url}/rates/historical") do |req|
+      req.headers["Authorization"] = "Bearer #{api_key}"
+      req.params["date"] = date.to_s
+      req.params["from"] = from
+      req.params["to"] = to
+    end
```

Comment:
hmm I think you can have som fun like:

```
def request url
  response = Faraday.get("#{base_url}/rates/historical") do |req|
      req.headers["Authorization"] = "Bearer #{api_key}"
      req.params = yield
   end 
   
   # ...
end

def fetch_exchange_rate(from:, to:, date:)
  request("#{base_url}/rates/historical") do
    { date: date.to_s, from: to: }
  end
end
```

Reply:
Yeah I've done this kinda thing in HTTP adapters before. It's dry-er. But I always ended up regretting it. It's easier to come back to all the request details being colocated.

Reply:
Until they get out of hand that is. This kind of cleanup is always a small effort away if needed.

Reply:
Fair! I would at least reduce the boiler plate with:

```
req.params = {
   date: 
}
```

Just makes it a smidge easier to modify (and read, I'd say).

Reply:
Moved some things around in https://github.com/maybe-finance/maybe/pull/561/commits/c49782ebfab9289bfd871c4616bd64e48fa063b8 and these classes ended up arguably messier.

I think that's fine. It's all self contained and easily refactored as patterns emerge within each provider.

Gut tells me to avoid sharing too much code between providers. Yes, they're all providers in the realm of Maybe. But they're each their own thing and have their own lifecycles. Probably best not to tie them together?